### ISSUE TITLE
Loss calculation error?

### ISSUE THREAD

**Issue by YongfeiYan** [dev4488/VAE_gumble_softmax#1]
Hi, there seems a redundant minus before the calculation of KLD.
`
KLD = torch.sum(qy * (log_qy - g)) # not  -torch.sum(qy * (log_qy - g))
`
See [this implementation](https://github.com/ericjang/gumbel-softmax/blob/master/Categorical%20VAE.ipynb).
I got better images after changing KLD. See [my modifications](https://github.com/YongfeiYan/Gumbel_Softmax_VAE)

**Comment by dev4488** [comment]
Thanks YongfeiYan! Yes, you are correct. I have edited it accordingly. 
Also, in your implementation, you should take *log* of the 1./categorical_dim term too.

Thanks again!
